src/start-server.ts

```diff
@@ -0,0 +1,12 @@
+import { ApolloServer } from 'apollo-server';
+import { AppDataSource, createDataSource } from './data-source';
+import { resolvers } from './graphql/resolvers';
+import { typeDefs } from './graphql/schema';
+
+export async function startServer() {
+  createDataSource();
+  AppDataSource.initialize();
```

Comment:
O `initialize` é uma promise, mas aqui está sem await. Tem que tomar cuidado porque essa função `startServer` pode acabar finalizando sem que o banco tenha sido conectado ainda.

Então se algum lugar no seu código for chamar o `startServer` de forma a esperar o término tanto da conexão do banco quanto o setup do server, pode ser que isso não aconteça.